platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkJavaCompiler.java

```diff
@@ -62,15 +67,20 @@ public WorkResult execute(JavaCompileSpec spec) {
         LOGGER.info("Compiling with JDK Java compiler API.");
 
         ApiCompilerResult result = new ApiCompilerResult();
-        JavaCompiler.CompilationTask task = createCompileTask(spec, result);
+        JavaCompiler.CompilationTask task;
+        try {
+            task = createCompileTask(spec, result);
+        } catch (RuntimeException ex) {
+            throw problemsService.getInternalReporter().rethrowing(ex, builder -> buildProblemFrom(ex, builder));
+        }
         boolean success = task.call();
         if (!success) {
             throw new CompilationFailedException(result);
         }
         return result;
     }
 
-    @SuppressWarnings("DefaultCharset")
+@SuppressWarnings("DefaultCharset")
```

Comment:
Accidental change, right?

```suggestion
    @SuppressWarnings("DefaultCharset")
```